### PR TITLE
solution

### PR DESCRIPTION
**Pull request by NikitaNikolaitsev**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1 +1,12 @@
-# write your code here
+def calculate_team_total_rating(players: list) -> int:
+    return sum([player.get_rating() for player in players])
+
+
+def elves_concert(elves: list) -> None:
```
Comment (DevSpaciX) [inline app/main.py:5]: It’s better to write exactly what you expect in the `list` . Please specify the correct typing

File: app/players/dwarves/dwarf.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from app.players.player import Player
+from abc import ABC
+
+
+class Dwarf(Player, ABC):
```
Comment (DevSpaciX) [inline app/players/dwarves/dwarf.py:5]: Do not repeat abstract methods from a parent if you have an abstract class that inherits another abstract class.
So `Player` already child of `ABC` class , so you can interhance just from `Player`

**Review by DevSpaciX — CHANGES_REQUESTED** [review]
Pls fix some mistakes in all places : )

**Comment by NikitaNikolaitsev** [comment]
I add 2 Question

File: app/players/dwarves/dwarf.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+from app.players.player import Player
+
+
+class Dwarf(Player):
+    # Why don't the tests
+    # throw errors if I haven't
+    # specified the abstract methods get_rating and player_info?
```
Comment (DevSpaciX) [inline app/players/dwarves/dwarf.py:7]: Good point , I will check it

File: app/main.py [hunk 4]
```diff
@@ -1 +1,17 @@
...
+def calculate_team_total_rating(players: list) -> int:
+    return sum([player.get_rating() for player in players])
+
+
+def elves_concert(elves: List[Elf]) -> None:  # correct?
+    for elf in elves:
+        elf.play_elf_song()
+
+
+def feast_of_the_dwarves(dwarfs: List[Dwarf]) -> None:  # correct?
```
Comment (DevSpaciX) [inline app/main.py:15]: Right !

**Review by DevSpaciX — APPROVED** [review]

**Pull request closed without merging**